pkg/storage/chunk/cache/cache.go

```diff
@@ -116,24 +101,11 @@ func New(cfg Config, reg prometheus.Registerer, logger log.Logger, cacheType sta
 	}
 
 	var caches []Cache
-
-	// Currently fifocache can be enabled in two ways.
-	// 1. cfg.EnableFifocache (old deprecated way)
-	// 2. cfg.EmbeddedCache.Enabled=true and cfg.EmbeddedCache.Distributed=false (new way)
-	if cfg.EnableFifoCache || cfg.EmbeddedCache.IsEnabled() {
-		var fifocfg FifoCacheConfig
-
-		if cfg.EnableFifoCache {
-			level.Warn(logger).Log("msg", "fifocache config is deprecated. use embedded-cache instead")
-			fifocfg = cfg.Fifocache
-		}
-
-		if cfg.EmbeddedCache.IsEnabled() {
-			fifocfg = FifoCacheConfig{
-				MaxSizeBytes:  fmt.Sprint(cfg.EmbeddedCache.MaxSizeMB * 1e6),
-				TTL:           cfg.EmbeddedCache.TTL,
-				PurgeInterval: cfg.EmbeddedCache.PurgeInterval,
-			}
+	if cfg.EmbeddedCache.IsEnabled() {
+		fifocfg := FifoCacheConfig{
```

Comment:
Shouldn't this be `EmbeddedCacheConfig`?

Reply:
Not always, it seems:
https://github.com/grafana/loki/blob/e5c425eef6980604d96ed5f341598b32e706b9e6/pkg/loki/modules.go#L587-L598

Things are getting pretty weird in this space. I think we need to standardise on a single, clear approach.
I don't think we want to keep groupcache at all

Reply:
``` but the former is now removed``` 
sorry correction. I meant to say groupcache is now removed.

Reply:
This is still a bit weird for me. `EmbeddedCacheConfig` and `FifoCacheConfig` are almost identical, so we should prefer `EmbeddedCacheConfig` to reduce complexity. We're nominally removing fifocache in this PR but by still using its configs and internal structs.

Maybe for another PR, but I think we should rename `FifoCache` to `EmbeddedCache` and simplify this part of the code.

Reply:
agree with this. pushed commits to remove/rename fifocache